keps/sig-network/4969-cluster-domain-pod-api/README.md

```diff
@@ -0,0 +1,690 @@
+
+Not applicable, as far as I can tell.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+Not applicable, as far as I can tell.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+Not applicable, as far as I can tell.
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No.
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+Enabling the feature should not result in any new API calls (it should be able
+to piggyback on the kubelet's existing status and DNS machinery).
+
+Consuming the feature over the REST API may require some clients to request
+their own Pod object during Pod startup.
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+The new field `.status.dns.clusterDomain` would be added to all Pod objects.
+
+For the default value of `cluster.local`, this would add 42 bytes to the 
+minified JSON serialization of each Pod.
+
+(`echo -n ',{"dns":{"clusterDomain":"cluster.local"}}' | wc -c`, assuming that
+`.status` is already set, but `.status.dns` is not.)
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+Beyond the size increase noted above, no.
+
+The information is already stored locally, and required to generate the
+`resolv.conf` file.
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No, beyond the Pod size increase already noted.
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
```

Comment:
I would encourage you to answer these in alpha.

Reply:
Since you're changing API server, the unavailability of etcd will affect the system, so not applicable isn't correct answer here.